### DISCUSSION TITLE
keep alive in settings missing?  or on purpose?

### DISCUSSION THREAD

**Discussion by timprepscius** [microsoft/msquic#1738] [General]
class MsQuicSettings : public QUIC_SETTINGS {

Should it have a:
SetKeepAliveIntervalMS or EnableKeepAliveWithIntervalMS?

**Comment by nibanks** [comment]
It has `KeepAliveIntervalMs`. Is that what you're looking for? A value of `0` means disabled.

**Reply by timprepscius** [reply]
Ah, I thought the MsQuicSettings would wrap it, like it does for other settings.

e.g.
 MsQuicSettings& SetMinimumMtu(uint16_t Mtu) { MinimumMtu = Mtu; IsSet.MinimumMtu = TRUE; return *this; }

**Reply by nibanks** [reply]
Ah, I wasn't paying attention to the fact you were talking about `MsQuicSettings`. We have just been adding helper functions to `MsQuicSettings` as necessary. Feel free to send a PR to add what you need (or go all out and add them all)!

**Reply by ThadHouse** [reply]
One thing to note about the C++ API is its not officially supported. It's possible they might change in the future without warning. It's mostly a test API. That being said its not likely to change since it works pretty well.
